import database1
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import QMessageBox

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")       
        MainWindow.resize(548, 479)
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        MainWindow.setFont(font)
        MainWindow.setStyleSheet("background-color: #c5d5cb;")
        MainWindow.setIconSize(QtCore.QSize(24, 24))
        MainWindow.setToolButtonStyle(QtCore.Qt.ToolButtonIconOnly)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(130, 60, 281, 51))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.label.setFont(font)
        self.label.setStyleSheet("background-color: #9fa8a3;")
        self.label.setObjectName("label")
        self.userbut = QtWidgets.QPushButton(self.centralwidget , clicked = lambda: self.users_window())
        self.userbut.setGeometry(QtCore.QRect(200, 170, 131, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.userbut.setFont(font)
        self.userbut.setObjectName("userbut")
        self.adminbut = QtWidgets.QPushButton(self.centralwidget , clicked = lambda: self.admin_window())
        self.adminbut.setGeometry(QtCore.QRect(200, 260, 131, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.adminbut.setFont(font)
        self.adminbut.setObjectName("adminbut")
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 548, 21))
        self.menubar.setObjectName("menubar")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "Voting System"))
        MainWindow.setWindowIcon(QtGui.QIcon("Myicon"))
        self.label.setText(_translate("MainWindow", "      Are you a User or Admin"))
        self.userbut.setText(_translate("MainWindow", "User"))
        self.adminbut.setText(_translate("MainWindow", "Admin"))

    def users_window(self):
        self.window = QtWidgets.QMainWindow() 
        ui = Ui_SecondWindow()
        ui.setupUi(self.window )
        self.window.show()
        MainWindow.hide()
    
    def admin_window(self):
        self.window1 = QtWidgets.QMainWindow()
        ui1 = Ui_ThirdWindow()
        ui1.setupUi(self.window1)
        self.window1.show()
        MainWindow.hide()

class Ui_SecondWindow(object):
    def setupUi(self, SecondWindow ):
        SecondWindow.setObjectName("SecondWindow")
        SecondWindow.resize(588, 437)
        SecondWindow.setStyleSheet("background-color:#c5d5cb;")
        self.centralwidget = QtWidgets.QWidget(SecondWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(160, 130, 81, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(150, 200, 101, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.userID = QtWidgets.QLineEdit(self.centralwidget)
        self.userID.setGeometry(QtCore.QRect(250, 140, 141, 31))
        self.userID.setFont(font)
        self.userID.setObjectName("userID")
        self.password = QtWidgets.QLineEdit(self.centralwidget)
        self.password.setGeometry(QtCore.QRect(250, 210, 141, 31))
        self.password.setFont(font)
        self.password.setObjectName("password")
        self.submit = QtWidgets.QPushButton(self.centralwidget)
        self.submit.setGeometry(QtCore.QRect(280, 310, 91, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.submit.setFont(font)
        self.submit.setObjectName("submit")
        SecondWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(SecondWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 588, 21))
        self.menubar.setObjectName("menubar")
        SecondWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(SecondWindow)
        self.statusbar.setObjectName("statusbar")
        SecondWindow.setStatusBar(self.statusbar)

        self.retranslateUi(SecondWindow)
        QtCore.QMetaObject.connectSlotsByName(SecondWindow)

    def retranslateUi(self, SecondWindow):
        _translate = QtCore.QCoreApplication.translate
        SecondWindow.setWindowTitle(_translate("SecondWindow", "User Page"))
        SecondWindow.setWindowIcon(QtGui.QIcon("Myicon"))
        self.label.setText(_translate("SecondWindow", "UserID:"))
        self.label_2.setText(_translate("SecondWindow", "Password:"))
        self.submit.setText(_translate("SecondWindow", "Login"))

class Ui_ThirdWindow(object):
    def setupUi(self, ThirdWindow):
        ThirdWindow.setObjectName("ThirdWindow")
        ThirdWindow.resize(588, 437)
        ThirdWindow.setStyleSheet("background-color:#c5d5cb;")
        self.centralwidget = QtWidgets.QWidget(ThirdWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(160, 130, 81, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(150, 200, 101, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.AdminID = QtWidgets.QLineEdit(self.centralwidget)
        self.AdminID.setGeometry(QtCore.QRect(250, 140, 141, 31))
        self.AdminID.setFont(font)
        self.AdminID.setObjectName("AdminID")
        self.Admin_password = QtWidgets.QLineEdit(self.centralwidget)
        self.Admin_password.setGeometry(QtCore.QRect(250, 210, 141, 31))
        self.Admin_password.setFont(font)
        self.Admin_password.setObjectName("Admin_password")
        self.submit = QtWidgets.QPushButton(self.centralwidget, clicked = lambda: self.validating_admin(ThirdWindow))
        self.submit.setGeometry(QtCore.QRect(280, 310, 91, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.submit.setFont(font)
        self.submit.setObjectName("submit")
        ThirdWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(ThirdWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 588, 21))
        self.menubar.setObjectName("menubar")
        ThirdWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(ThirdWindow)
        self.statusbar.setObjectName("statusbar")
        ThirdWindow.setStatusBar(self.statusbar)
        self.retranslateUi(ThirdWindow)
        QtCore.QMetaObject.connectSlotsByName(ThirdWindow)

    def retranslateUi(self, ThirdWindow):
        _translate = QtCore.QCoreApplication.translate
        ThirdWindow.setWindowTitle(_translate("ThirdWindow", "Admin Page"))
        ThirdWindow.setWindowIcon(QtGui.QIcon("Myicon"))
        self.label.setText(_translate("ThirdWindow", "AdminID:"))
        self.label_2.setText(_translate("ThirdWindow", "Password:"))
        self.submit.setText(_translate("ThirdWindow", "Login"))
    
    def validating_admin(self , ThirdWindow):
        AdminID = self.AdminID.text()
        print(AdminID)
        Admin_password = self.Admin_password.text()
        print(type(Admin_password))
        validation = f"""
        select password from admin where adminID = {AdminID};"""
        result = database1.execute_query(None , validation)
        if (result == Admin_password) :
            print("Correct password")  
            self.window2 = QtWidgets.QMainWindow()
            ui2 = Ui_Admin_main_page()
            ui2.setupUi(self.window2 , AdminID)
            self.window2.show()
            print(result)
            ThirdWindow.hide() 
        elif (result == None) :
            msg = QMessageBox() 
            msg.setIcon(QMessageBox.Critical)
            msg.setText("    Wrong UserID or Password           ")
            msg.setWindowTitle("Alert MessageBox")
            msg.setWindowIcon(QtGui.QIcon("password"))
            msg.setStandardButtons(QMessageBox.Ok | QMessageBox.Cancel)
            retval = msg.exec_()
        else :
            msg = QMessageBox() 
            msg.setIcon(QMessageBox.Critical)
            msg.setText("    Wrong UserID or Password           ")
            msg.setWindowTitle("Alert MessageBox")
            msg.setWindowIcon(QtGui.QIcon("password"))
            msg.setStandardButtons(QMessageBox.Ok | QMessageBox.Cancel)
            retval = msg.exec_()            

class Ui_Admin_main_page(object):
    def setupUi(self, Admin_main_page ,AdminID):
        Admin_main_page.setObjectName("Admin_main_page")
        Admin_main_page.resize(588, 437)
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        Admin_main_page.setFont(font)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("Myicon.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        Admin_main_page.setWindowIcon(icon)
        Admin_main_page.setStyleSheet("background-color:#c5d5cb;")
        Admin_main_page.setIconSize(QtCore.QSize(23, 23))
        self.centralwidget = QtWidgets.QWidget(Admin_main_page)
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.centralwidget.setFont(font)
        self.centralwidget.setObjectName("centralwidget")
        self.Creating_User = QtWidgets.QPushButton(self.centralwidget, clicked = lambda: self.creating_user_page(Admin_main_page))
        self.Creating_User.setGeometry(QtCore.QRect(230, 130, 131, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.Creating_User.setFont(font)
        self.Creating_User.setObjectName("Creating_User")
        self.Creating_Campaign = QtWidgets.QPushButton(self.centralwidget)
        self.Creating_Campaign.setGeometry(QtCore.QRect(220, 210, 161, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.Creating_Campaign.setFont(font)
        self.Creating_Campaign.setObjectName("Creating_Campaign")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(170, 60, 241, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.openGLWidget = QtWidgets.QOpenGLWidget(self.centralwidget)
        self.openGLWidget.setGeometry(QtCore.QRect(-300, 330, 300, 200))
        self.openGLWidget.setObjectName("openGLWidget")
        self.logoutButton = QtWidgets.QCommandLinkButton(self.centralwidget)
        self.logoutButton.setGeometry(QtCore.QRect(450, 0, 111, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(11)
        self.logoutButton.setFont(font)
        self.logoutButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("logout.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.logoutButton.setIcon(icon1)
        self.logoutButton.setIconSize(QtCore.QSize(24, 24))
        self.logoutButton.setObjectName("logoutButton")
        self.Check_Result = QtWidgets.QPushButton(self.centralwidget)
        self.Check_Result.setGeometry(QtCore.QRect(230, 290, 131, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.Check_Result.setFont(font)
        self.Check_Result.setObjectName("Check_Result")
        self.Admin_label = QtWidgets.QLabel(self.centralwidget)
        self.Admin_label.setGeometry(QtCore.QRect(10, 0, 181, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(11)
        self.Admin_label.setFont(font)
        self.Admin_label.setText(f"Welcome, Admin{AdminID}")
        self.Admin_label.setObjectName("Admin_label")
        Admin_main_page.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(Admin_main_page)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 588, 21))
        self.menubar.setObjectName("menubar")
        Admin_main_page.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(Admin_main_page)
        self.statusbar.setObjectName("statusbar")
        Admin_main_page.setStatusBar(self.statusbar)

        self.retranslateUi(Admin_main_page)
        QtCore.QMetaObject.connectSlotsByName(Admin_main_page)

    def retranslateUi(self, Admin_main_page):
        _translate = QtCore.QCoreApplication.translate
        Admin_main_page.setWindowTitle(_translate("Admin_main_page", "Admin Panel"))
        self.Creating_User.setText(_translate("Admin_main_page", "Create User"))
        self.Creating_Campaign.setText(_translate("Admin_main_page", "Create Campaign"))
        self.label.setText(_translate("Admin_main_page", "What you would like to do?"))
        self.logoutButton.setText(_translate("Admin_main_page", "Logout"))
        self.Check_Result.setText(_translate("Admin_main_page", "Check Result"))
    
    def creating_user_page(self ,Admin_main_page):
        self.window3 = QtWidgets.QMainWindow()
        ui3 = Ui_Admin_creating_user()
        ui3.setupUi(self.window3)
        self.window3.show()
        Admin_main_page.hide()
        


class Ui_Admin_creating_user(object):
    def setupUi(self, Admin_creating_user):
        Admin_creating_user.setObjectName("Admin_creating_user")
        Admin_creating_user.resize(633, 466)
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        Admin_creating_user.setFont(font)
        Admin_creating_user.setWindowTitle("Creating User")
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("Myicon.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        Admin_creating_user.setWindowIcon(icon)
        Admin_creating_user.setStyleSheet("background-color:#c5d5cb;")
        Admin_creating_user.setIconSize(QtCore.QSize(57, 29))
        Admin_creating_user.setTabShape(QtWidgets.QTabWidget.Triangular)
        self.centralwidget = QtWidgets.QWidget(Admin_creating_user)
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(14)
        self.centralwidget.setFont(font)
        self.centralwidget.setObjectName("centralwidget")
        self.openGLWidget = QtWidgets.QOpenGLWidget(self.centralwidget)
        self.openGLWidget.setGeometry(QtCore.QRect(-300, 330, 300, 200))
        self.openGLWidget.setObjectName("openGLWidget")
        self.logoutButton = QtWidgets.QCommandLinkButton(self.centralwidget)
        self.logoutButton.setGeometry(QtCore.QRect(530, 0, 111, 41))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(11)
        self.logoutButton.setFont(font)
        self.logoutButton.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("logout.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.logoutButton.setIcon(icon1)
        self.logoutButton.setIconSize(QtCore.QSize(24, 24))
        self.logoutButton.setObjectName("logoutButton")
        self.back = QtWidgets.QCommandLinkButton(self.centralwidget)
        self.back.setGeometry(QtCore.QRect(0, -10, 61, 51))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(14)
        self.back.setFont(font)
        self.back.setText("")
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("icons8-left-arrow-64.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.back.setIcon(icon2)
        self.back.setIconSize(QtCore.QSize(40, 40))
        self.back.setObjectName("back")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(110, 70, 131, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(110, 200, 91, 21))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.label_3 = QtWidgets.QLabel(self.centralwidget)
        self.label_3.setGeometry(QtCore.QRect(110, 130, 171, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.full_name = QtWidgets.QLineEdit(self.centralwidget)
        self.full_name.setGeometry(QtCore.QRect(290, 70, 201, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.full_name.setFont(font)
        self.full_name.setText("")
        self.full_name.setObjectName("full_name")
        self.father_name = QtWidgets.QLineEdit(self.centralwidget)
        self.father_name.setGeometry(QtCore.QRect(290, 130, 201, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.father_name.setFont(font)
        self.father_name.setObjectName("father_name")
        self.voter_id = QtWidgets.QLineEdit(self.centralwidget)
        self.voter_id.setGeometry(QtCore.QRect(290, 190, 131, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.voter_id.setFont(font)
        self.voter_id.setObjectName("voter_id")
        self.date_of_birth = QtWidgets.QDateEdit(self.centralwidget)
        self.date_of_birth.setGeometry(QtCore.QRect(290, 260, 121, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(11)
        self.date_of_birth.setFont(font)
        self.date_of_birth.setObjectName("date_of_birth")
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        self.label_4.setGeometry(QtCore.QRect(110, 260, 141, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setGeometry(QtCore.QRect(480, 360, 91, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.pushButton.setFont(font)
        self.pushButton.setObjectName("pushButton")
        self.pushButton_2 = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_2.setGeometry(QtCore.QRect(320, 360, 91, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.pushButton_2.setFont(font)
        self.pushButton_2.setObjectName("pushButton_2")
        Admin_creating_user.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(Admin_creating_user)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 633, 21))
        self.menubar.setObjectName("menubar")
        Admin_creating_user.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(Admin_creating_user)
        self.statusbar.setObjectName("statusbar")
        Admin_creating_user.setStatusBar(self.statusbar)

        self.retranslateUi(Admin_creating_user)
        QtCore.QMetaObject.connectSlotsByName(Admin_creating_user)

    def retranslateUi(self, Admin_creating_user):
        _translate = QtCore.QCoreApplication.translate
        self.logoutButton.setText(_translate("Admin_creating_user", "Logout"))
        self.label.setText(_translate("Admin_creating_user", "Full Name"))
        self.label_2.setText(_translate("Admin_creating_user", "Voter ID"))
        self.label_3.setText(_translate("Admin_creating_user", "Father Name"))
        self.label_4.setText(_translate("Admin_creating_user", "Date of Birth"))
        self.pushButton.setText(_translate("Admin_creating_user", "Create"))
        self.pushButton_2.setText(_translate("Admin_creating_user", "Cancel"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())
    